### PR TITLE
ECER-1918: Previous Name changes

### PR DESCRIPTION
**Pull request by pcatkins**
Failed merge for stories/ECER-1918

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/ProfileEndpoints.cs [hunk 1]
```diff
@@ -34,18 +35,38 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
...
+  public string Phone { get; set; } = null!;
+  public Address? ResidentialAddress { get; set; }
+  public Address? MailingAddress { get; set; }
+  public IEnumerable<PreviousName> PreviousNames { get; set; } = Array.Empty<PreviousName>();
+};
+
+/// <summary>
+/// Previous Name
+/// </summary>
+public record PreviousName ([Required] string? FirstName, [Required] string? LastName)
```
Comment (farzadnadiri) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/ProfileEndpoints.cs:55]: if these fields are required why do we have them nullable?

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/user.ts [hunk 2]
```diff
@@ -8,16 +8,17 @@ export interface UserState {
...
+  persist: true,
   state: (): UserState => ({
     userInfo: null,
     userProfile: null,
   }),
   getters: {
     hasUserInfo: (state): boolean => state.userInfo !== null,
     hasUserProfile: (state): boolean => state.userProfile !== null,
+    unverifiedPreviousNames: (state): Components.Schemas.PreviousName[] => {
+      return state.userProfile?.previousNames?.filter((name) => name.status == "Unverified") ?? [];
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/user.ts:20]: super small nitpik. 

best to use === for equality checks in javascript.

**Review by SoLetsDev — COMMENTED** [review]
still reviewing small nitpik

File: src/ECER.Tests/Integration/RegistryApi/ProfileTests.cs [hunk 3]
```diff
@@ -33,6 +34,15 @@ public async Task PutProfile_Existing_Returned()
     });
   }
 
+  private PreviousName CreatePreviousName()
+  {
+    var faker = new Faker("en_CA");
```
Comment (SoLetsDev) [inline src/ECER.Tests/Integration/RegistryApi/ProfileTests.cs:39]: confused. doesn't look like we are using faker here?

**Review by farzadnadiri — APPROVED** [review]

**Review by SoLetsDev — APPROVED** [review]

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/ProfileEndpoints.cs (hunk 1 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/ProfileEndpoints.cs:55]: Will clean up in next previous name branch

**Pull request merged**